### DISCUSSION TITLE
Show warning on tab close

### DISCUSSION THREAD

**Discussion by meslahik** [zauberzeug/nicegui#2742] [Q&A]
### Question

Is there a simple way to show users a warning before a tab is closed?

**Comment by falkoschindler** [comment]
Something like this? https://github.com/zauberzeug/nicegui/discussions/1700 🙂

**Comment by meslahik** [comment]
Thank you @falkoschindler! 😀

There are scenarios that showing warning on a page not make sense after some action!
To disable this warning, one can run javascript code using NiceGUI method as:
```python
ui.run_javascript('window.onbeforeunload = null;')
```
